### PR TITLE
SP2-814: Added create SAN endpoint for co-ordinator

### PR DESCRIPTION
**Pull request by pmcphee77**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentService.kt [hunk 1]
```diff
@@ -26,6 +32,18 @@ class AssessmentService(
...
+  fun createAndAudit(userDetails: UserDetails): CreateAssessmentResponse {
+    return CreateAssessmentResponse.from(
+      create().also {
+        AssessmentVersionAudit(
+          assessmentVersion = it.assessmentVersions.first(),
+          userDetails = userDetails,
+        ).run(assessmentVersionAuditRepository::save)
+      },
+    )
+  }
```
Comment (devkokov) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentService.kt:45]:
```suggestion
  fun createAndAudit(userDetails: UserDetails): Assessment {
    return create().also {
      AssessmentVersionAudit(
        assessmentVersion = it.assessmentVersions.first(),
        userDetails = userDetails,
      ).run(assessmentVersionAuditRepository::save)
    }
  }
```

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/controller/AssessmentController.kt [hunk 2]
```diff
@@ -79,4 +82,18 @@ class AssessmentController(
...
+  @ApiResponses(
+    value = [
+      ApiResponse(responseCode = "201", description = "Assessment created"),
+    ],
+  )
+  @PreAuthorize("hasRole('ROLE_STRENGTHS_AND_NEEDS_WRITE')")
+  fun createAssessment(
+    @RequestBody
+    userDetails: UserDetails,
+  ) = assessmentService.createAndAudit(userDetails)
```
Comment (devkokov) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/controller/AssessmentController.kt:98]:
```suggestion
  ) = assessmentService.createAndAudit(userDetails).run(CreateAssessmentResponse::from)
```

Comment (pmcphee77) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/controller/AssessmentController.kt]: Ah yes. Sorry - not used to returning entities to the controller. But it should be consistent! Changed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentService.kt (hunk 1 above)
Comment (pmcphee77) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentService.kt]: Done

Comment (devkokov) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentService.kt]: Fab, thanks!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/controller/AssessmentController.kt (hunk 2 above)
Comment (devkokov) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/controller/AssessmentController.kt]: Thanks!

**Review by devkokov — APPROVED** [review]

**Pull request merged**